include/arch/arm/arch/64/mode/kernel/vspace.h

```diff
@@ -86,14 +67,22 @@ static const region_t BOOT_RODATA *mode_reserved_region = NULL;
 
 static inline exception_t performASIDPoolInvocation(asid_t asid, asid_pool_t *poolPtr, cte_t *cte)
 {
+    asid_map_t asid_map;
     cap_page_upper_directory_cap_ptr_set_capPUDMappedASID(&cte->cap, asid);
     cap_page_upper_directory_cap_ptr_set_capPUDIsMapped(&cte->cap, 1);
-    poolPtr->array[asid & MASK(asidLowBits)] =
-        PUDE_PTR(cap_page_upper_directory_cap_get_capPUDBasePtr(cte->cap));
+    asid_map = asid_map_asid_map_vspace_new(
```

Comment:
nitpicking: go for c99?
```suggestion
    asid_map_t asid_map = asid_map_asid_map_vspace_new(
```